ftlib/impl.py

```diff
@@ -0,0 +1,225 @@
+import logging
+import threading
+
+from ftlib.commlib.commlib_status import CommLibStatus
+from ftlib.consensus.consensus_status import ConsensusMode, ConsensusStatus
+from ftlib.ftlib_status import FTAllReduceStatus, FTRebuildStatus
+from ftlib.rank_assign_scheme import get_rank_size
+
+# in the ftlib package, user is able to initialize the package
+# with specific consensus method and communication library
+
+
+def gen_constant_class(implemented_list):
+    class SOMEList:
+        IMPLEMENTED = implemented_list
+
+        @staticmethod
+        def contains(item):
+            return item in SOMEList.IMPLEMENTED
+
+    return SOMEList
+
+
+CommLibList = gen_constant_class(["dummy_NCCL", "pytorch"])
+ConsensusList = gen_constant_class(["gossip", "shared_storage"])
```

Comment:
The above constants are only internal? If so, prefix the variable names with underscore